libcaf_core/caf/delegate.test.cpp

```diff
@@ -0,0 +1,109 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/test/caf_test_main.hpp"
+#include "caf/test/fixture/deterministic.hpp"
+#include "caf/test/test.hpp"
+
+#include "caf/event_based_actor.hpp"
+#include "caf/scoped_actor.hpp"
+#include "caf/sec.hpp"
+#include "caf/send.hpp"
+
+namespace fixture = caf::test::fixture;
+
+WITH_FIXTURE(fixture::deterministic) {
+
+#ifdef CAF_ENABLE_EXCEPTIONS
+TEST("delegates pass on the response properly to the sender") {
+  using namespace caf;
+  auto count = std::make_shared<int32_t>(0);
+  auto worker = sys.spawn([count] {
+    return behavior{
+      [count](int32_t value) {
+        *count = *count + value;
+        return *count;
+      },
+    };
+  });
+  auto delegator = sys.spawn(
+    [](event_based_actor* self, caf::actor worker) {
+      return behavior{
+        [self, worker](int32_t x) {
+          auto promise = self->make_response_promise<int>();
+          return promise.delegate(worker, x);
```

Comment:
This certainly works, but it's quite verbose. Why go through a promise?

```suggestion
          return self->delegate(worker, x);
```